src/environment.rs

```diff
@@ -120,6 +120,16 @@ fn normalize_path(path: path::PathBuf) -> Result<path::PathBuf, Error> {
     Ok(res)
 }
 
+/// Safe wrapper for the getuid function
+fn getuid() -> libc::uid_t {
+    unsafe { libc::getuid() }
+}
```

Comment:
https://docs.rs/nix/0.9.0/nix/unistd/fn.getuid.html?

Reply:
I thought about it, but decided against the additional dependency given the wrapper is tiny. I could be persuaded.

Reply:
I mean, you're adding in an extra dependency on libc already - probably is nicer to use `nix` here.

Reply:
So `libc` is already a transitive dependency, so adding it is just exposes it. `nix` depends on `libc` also, and re-exports it (it's `Uid` and `Gid` types are structures over `libc`s `uid_t` and `gid_t`).

However, I think it might be nicer to have the typed wrappers to `Uid` and `Gid` (rather than type aliases for `u32`), which `nix` does already. I'll give it a go tonight. If it works naturally I'll keep it.

Reply:
Thanks!

Reply:
Turned out to be pretty clean, so I pushed the commit.